### ISSUE TITLE
What brought you to JavaScript and what tips/hacks do you have for beginners wanting to speed up their learning?

### ISSUE THREAD

**Issue by luclemo** [wesbos/ama#37]
What brought you to start exploring/using JavaScript, and can you describe the path for your own learning? How long before you felt you had enough of a solid foundation to build on, that moment where you understand how all the moving parts work and you can start to get efficient/creative rather than trying to figure out what's happening?

Do you have tips for beginners determined to learn, but who don't have a programming background nor much opportunity to use it in their day-to-day projects? How can a beginner find/create small learning projects which are not too complex when they don't really know what "too complex" looks like?

Hope that's not too long!

**Comment by wesbos** [comment]
Whew - great question. TLDR: Build lots of stuff! 

> What brought you to start exploring/using JavaScript, and can you describe the path for your own learning?

I stumbled upon the jQuery community, primarily the #jquery IRC chatroom and they were extremely helpful in fixing my problems but also becoming friends. I really liked the community but I _hated_ JavaScript at first and can very clearly remember contemplating just dropping it all together. But I powered through and now I absolutely love it. 

> How long before you felt you had enough of a solid foundation to build on, that moment where you understand how all the moving parts work and you can start to get efficient/creative rather than trying to figure out what's happening?

It was a good 3-4 years before I felt really comfortable with JavaScript, but at that time JavaScript wasn't really such a big part of websites as it is today. I don't think I ever had a an ah-ha moment and I think I'm still learning a lot about JavaScript - I always will be. The more I learn, the more I realize I can still learn. 

> Do you have tips for beginners determined to learn, but who don't have a programming background nor much opportunity to use it in their day-to-day projects? 

Build 1,000 little things. Keep building stuff and you will get amazing. Don't worry about it being 100% perfect or following best practices - that is what refactoring is for - but just make something so that it works. I've got tons and tons of little "things" I've made on my computer - some won't see the light of day but I learned a ton along the way!

> How can a beginner find/create small learning projects which are not too complex when they don't really know what "too complex" looks like?

Whenever you see a cool effect on a website, try and build just that little piece. Whenever you have an idea on how to manipulate data, try it!  I think I might have a little bit of a creative mind because these types of things pop into my head on a daily basis, and I love to take 20-30 mins out of my day to work on them -  maybe I should create a list of them... 


Thanks for the Q :+1:
